tsdb/engine/tsm1/engine.go

```diff
@@ -256,9 +214,9 @@ func NewEngine(id uint64, idx tsdb.Index, path string, walPath string, sfile *ts
 		formatFileName:                DefaultFormatFileName,
 		stats:                         stats,
 		compactionLimiter:             opt.CompactionLimiter,
-		scheduler:                     newScheduler(stats, opt.CompactionLimiter.Capacity()),
 		seriesIDSets:                  opt.SeriesIDSets,
 	}
+	e.scheduler = newScheduler(&e.activeCompactions, opt.CompactionLimiter.Capacity())
```

Comment:
Can the `newScheduler` call not still be part of the struct initialization? Seems like the first argument could have been updated rather than putting it on a new line.

Reply:
That would be a compiler error - the engine gives a pointer to a member of itself to `newScheduler`, which you can't do until after instantiation.

Reply:
Ah of course, thanks!

Reply:
But, it's probably also a code smell, will fix properly